extensions/expression-language/CMakeLists.txt

```diff
@@ -96,20 +96,22 @@ find_package(FLEX REQUIRED)
 bison_target(
     el-parser
     ${CMAKE_CURRENT_SOURCE_DIR}/Parser.yy
-    ${CMAKE_CURRENT_SOURCE_DIR}/Parser.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/el-parser/Parser.cpp
 )
 
 flex_target(
     el-scanner
     ${CMAKE_CURRENT_SOURCE_DIR}/Scanner.ll
-    ${CMAKE_CURRENT_SOURCE_DIR}/Scanner.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/el-scanner/Scanner.cpp
     COMPILE_FLAGS --c++
 )
 
 add_flex_bison_dependency(el-scanner el-parser)
 
 include_directories(./ ../../libminifi/include  ../../libminifi/include/core)
 include_directories(SYSTEM ../../thirdparty/)
+include_directories(SYSTEM el-parser)
+include_directories(SYSTEM el-scanner)
```

Comment:
What warnings are fixed here? I'd rather not add any more directory-based include paths unless absolutely necessary.

Reply:
Previously we generated the parser and scanner header and source files into the `extensions/expression-language/` directory and the files in that directory were compiled with the flags handling warnings as errors. Because of this when some warnings were found in the generated header files included by our own source files that resulted in a compile error. Moving the generated files to separate include directories and marking them as system includes ignores the warnings in those header files.

Reply:
Ok, got it. But I still thing that we should not add directory-based include paths, and add them to each target as necessary. It would also be nicer to generate these files under the build directory, instead of the source tree.

Reply:
Updated in be9cace2da1aa161129ac9c22986033dafc199db